core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -149,12 +149,25 @@ class LogCleaner(val config: CleanerConfig,
   }
 
   /**
-   * For testing, a way to know when work has completed. This method blocks until the
+   * For testing, a way to know when work has completed. This method waits until the
    * cleaner has processed up to the given offset on the specified topic/partition
+   *
+   * @param topic The Topic to be cleaned
+   * @param part The partition of the topic to be cleaned
+   * @param offset The first dirty offset that the cleaner doesn't have to clean
+   * @param maxWaitMs The maximum time in ms to wait for cleaner
+   *
+   * @return A boolean indicating whether the work has completed before timeout
    */
-  def awaitCleaned(topic: String, part: Int, offset: Long, timeout: Long = 30000L): Unit = {
-    while(!cleanerManager.allCleanerCheckpoints.contains(TopicAndPartition(topic, part)))
-      Thread.sleep(10)
+  def awaitCleaned(topic: String, part: Int, offset: Long, maxWaitMs: Long = 60000L): Boolean = {
```

Comment:
Thanks for implementing the timeout. I, personally, prefer the following implementation:

```scala
def awaitCleaned(topic: String, part: Int, offset: Long, maxWaitMs: Long = 60000L): Boolean = {
  def isCleaned = cleanerManager.allCleanerCheckpoints.get(TopicAndPartition(topic, part)).fold(false)(_ >= offset)
  var remainingWaitMs = maxWaitMs
  while (!isCleaned && remainingWaitMs > 0) {
    val sleepTime = math.min(100, remainingWaitMs)
    Thread.sleep(sleepTime)
    remainingWaitMs -= sleepTime
  }
  isCleaned
```

It avoids `return`, which is discouraged in Scala, makes the loop condition clearer and the timeout only controls when to exit the loop, the return value is based on `isCleaned` only.